### DISCUSSION TITLE
Plugin says function doesn't have a snake case name, but it gets annotated with #[allow(non_snake_case)]

### DISCUSSION THREAD

**Discussion by byte-sized-emi** [intellij-rust/intellij-rust#9917] [IntelliJ Rust Plugin Development]
Basically, I have this code:

```rust
#[some_macro]
fn nonSnakeCaseName() { }
```

some_macro adds the `#[allow(non_snake_case)]` annotation, which clippy for example correctly uses and therefore doesn't display an error, but the plugin tells me that it should be called non_snake_case_name.

**Comment by Undin** [comment]
I've created a separate issue for this problem #9918
